### PR TITLE
[RFR] Draw a Labyrinth board

### PR DESCRIPTION
**Pull request by pemoreau**
This corresponds to this [Trello card](https://trello.com/c/sVpdoV4c/1-as-robert-i-can-launch-an-a-maze-ing-game)

- [x] draw a tile
- [x] draw several tiles
- [x] get a json board from labyrinth-api
- [x] draw a board

for next PR:
- [ ] introduce a Board react component
- [ ] test the generated html

screen shot 2018-10-17 at 15 34 28

### CODE REVIEW

File: src/Tile.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+
+const Tile = ({ x, y, degrees, matrix }) => (
+    <div class="tile-grid" style={{ gridColumn: x, gridRow: y, transform: `rotate(${degrees}deg)` }}>
```
[the remark this answers is not in the document]
Comment (djhi) [reply src/Tile.js]:
```jsx
const styles = { gridColumn: x, gridRow: y, transform: `rotate(${degrees}deg)` };

const Tile = ({ x, y, degrees, matrix }) => (
    <div class="tile-grid" style={styles}>
        ...
    </div>
);
```

[the remark this answers is not in the document]
Comment (djhi) [reply src/Tile.js]: nvm I missed the coordinates :tired_face:

File: src/App.js [hunk 2]
```diff
@@ -1,10 +1,35 @@
 import React, { Component } from 'react';
+import Tile from './Tile';
+import './App.css';
+import { createGame } from './common/game';
+import { flattenBoard } from './common/board';
+
+import { convertBoardXToDisplayX, convertBoardYToDisplayY } from './common/utils';
 
 class App extends Component {
+    constructor(props) {
```
Comment (jdemangeon) [inline src/App.js]: state = {
   game: createGame(),
}

File: src/common/board.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    { x: 5, y: 7 },
+    { x: 3, y: 7 },
+    { x: 1, y: 7 },
+    { x: -1, y: 5 },
+    { x: -1, y: 3 },
+    { x: -1, y: 1 },
+]);
+
+export const createEmptyBoard = () =>
+    produce({}, () => Array.from({ length: BOARD_SIZE }, () => Array.from({ length: BOARD_SIZE }, () => 0)));
```
Comment (jdemangeon) [inline src/common/board.js]: Array(10).fill(Array(10).fill(0))

File: src/common/game.js [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+import {
+    isCurrentTargetReached,
+    removeTargetCardToPlay,
+    movePlayerTo,
+    moveAllPlayers,
+    putPlayersBackOnBoard,
+} from './player';
+
+import { STATE } from './constants';
+
```
Comment (jdemangeon) [inline src/common/game.js]: Useless line

File: src/common/game.js [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+import { initGame } from './gameFactory';
+
+const NB_PLAYER = 1;
+const NB_TARGET_CARD = 24;
+
+export const createGame = () => {
+    let { board, players, remainingPathCard } = initGame(NB_PLAYER, NB_TARGET_CARD);
```
Comment (jdemangeon) [inline src/common/game.js]: Can you add spaces between lines to make it more readable?

File: src/common/game.js [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+    produce(game, draft => {
+        draft.state = STATE.END;
+    });
+
+const toNextPlayerTurn = game =>
+    produce(game, draft => {
+        draft.currentPlayerIndex = (draft.currentPlayerIndex + 1) % draft.players.length;
+    });
+
+const increasePlayerScoreIfOnTarget = game => {
```
Comment (jdemangeon) [inline src/common/game.js]: Same

File: src/common/game.js [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+            }
+        });
+    }
+    return result;
+};
+
+const computeImmediateReachablePositionsFromXY = (board, x, y) => {
+    const res = [];
+    const directions = Object.values(Direction);
+    directions.forEach(direction => {
```
Comment (jdemangeon) [inline src/common/game.js]: When you have time

return directions.reduce((res, direction) => {
if (...) {
res.push(...);
}
return res;
}, [])

File: src/common/game.spec.js [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+
+    it('should start with a not null currentPlayerReachablePositions', () => {
+        expect(currentPlayerReachablePositions).not.toBeNull();
+    });
+
+    it('should not be on the target', () => {
+        expect(playerX === targetX && playerY === targetY).toBeFalsy();
+    });
+
+    it('should increase score when target is reached', () => {
```
Comment (jdemangeon) [inline src/common/game.spec.js]: Can you skip it if it's not used ?

File: src/common/pathCard.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
+import { produce } from 'immer';
+
+export const Type = Object.freeze({ STRAIGHT: '┃', CORNER: '┗', CROSS: '┻' }); // pointing NORTH
```
Comment (jdemangeon) [inline src/common/pathCard.js]: Please use this const in react (in the mapping)
[Type.STRAIGHT]: ...

**Comment by jdemangeon** [comment]
I've not reviewed common files for the moment because they comes from the last project. PLease add tests to it in the future

**Pull request merged**